src/processes.c

```diff
@@ -390,25 +410,19 @@ static int ps_list_match (const char *name, const char *cmdline, procstat_t *ps)
 	return (0);
 } /* int ps_list_match */
 
-static void ps_update_counter (_Bool init, derive_t *group_counter,
-				derive_t *curr_counter, unsigned long *curr_value,
-				derive_t new_counter, unsigned long new_value)
+static void ps_update_counter (derive_t *group_counter, derive_t *curr_counter, 
+															 derive_t new_counter)
 {
-	if (init)
-	{
-		*curr_value = new_value;
-		*curr_counter += new_value;
-		*group_counter += new_value;
-		return;
-	}
+
+	unsigned long curr_value;
```

Comment:
Can we add some comment here to briefly mention the reason behind this adjustment? Just for future developer's benefit.

Reply:
I get concerned adding a comment about a change, when the change is not
visible once the code has been submitted. I'd rather make the comment in
the merged PR, WDYT?

Reply:
I think Ling meant adding a code comment in a commit that's part of this PR.

Reply:
?? I meant to add a comment in the code to point out that we are doing this adjustment because int32 can overflow.